File: bench/tasks/13-temperatures/python/main.py
temps = [22, 18, 25, 31, 29, 17, 24, 28, 20, 26]
total = hot = mild = cold = 0
for t in temps:
    total += t
    if t >= 28:
        hot += 1
    elif t >= 20:
        mild += 1
    else:
        cold += 1
avg = total // len(temps)
assert hot + mild + cold == len(temps) and avg == 24, "invalid temperature report"
print("Days measured: " + str(len(temps)))
print("Average: " + str(avg))
print("Hot days (28+): " + str(hot))
print("Mild days: " + str(mild))
print("Cold days (<20): " + str(cold))

File: bench/tasks/13-temperatures/go/main.go
package main

import "fmt"

func main() {
	temps := []int{22, 18, 25, 31, 29, 17, 24, 28, 20, 26}
	total, hot, mild, cold := 0, 0, 0, 0
	for _, t := range temps {
		total += t
		if t >= 28 {
			hot++
		} else if t >= 20 {
			mild++
		} else {
			cold++
		}
	}
	avg := total / len(temps)
	if hot+mild+cold != len(temps) || avg != 24 {
		panic("invalid temperature report")
	}
	fmt.Println("Days measured: " + fmt.Sprint(len(temps)))
	fmt.Println("Average: " + fmt.Sprint(avg))
	fmt.Println("Hot days (28+): " + fmt.Sprint(hot))
	fmt.Println("Mild days: " + fmt.Sprint(mild))
	fmt.Println("Cold days (<20): " + fmt.Sprint(cold))
}
